### PR TITLE
[Iguazio] Fix mount v3io for kfp jobs

### PR DESCRIPTION
**Pull request by alonmr**
https://jira.iguazeng.com/browse/ML-3392
It is not clear why this bug happens but it seems that there is a problem with nested methods inside a private function.
To get around it, I moved the nested function under a public one.

### CODE REVIEW

File: tests/system/test_kfp.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    @pytest.mark.enterprise
+    def test_kfp_with_mount(self):
+        code_path = str(self.assets_path / "kfp_with_mount.py")
+        kfp_with_v3io_mount = mlrun.code_to_function(
+            name="my-kfp-with-mount",
+            kind="job",
+            filename=code_path,
+            project=self.project_name,
+            image="mlrun/mlrun",
```
Comment (liranbg) [inline tests/system/test_kfp.py:37]: can be re removed, no?

Comment (alonmr) [reply tests/system/test_kfp.py:37]: looks like it is not being enriched so needs to be set explicitly

Comment (liranbg) [reply tests/system/test_kfp.py:37]: should be enriched server side

Comment (alonmr) [reply tests/system/test_kfp.py:37]: image is being enriched when `auto_build` option is enabled but it is currently not available with pipelines. will add it in a follow-up PR.

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**